packages/core/src/__tests__/dispatcher-real-ipfs.test.ts

```diff
@@ -86,7 +87,8 @@ describe('Dispatcher with real ipfs over http', () => {
     // shutdownController successfully interrupted waiting on IPFS.
 
     const getPromise = dispatcher.retrieveCommit(FAKE_CID)
+    await delay(300) // Apparently, with js-ipfs, we need some extra time.
```

Comment:
what happens if this delay isn't there?  It really shouldn't matter even if the shutdownController was triggered entirely before the call to retrieveCommit so I'd like to understand this better

Reply:
While it should not, in reality, it does. When testing with js-ipfs, signal is triggered before the `dag.get` runs. It seems, that whatever is inside `dag.get` does not respect already triggered abort signal. The delay makes sure that `dag.get` starts first, and only then the signal is triggered.

Reply:
so what happens?  Awaiting the `getPromise` hangs and the test times out?

Reply:
Hey @stbrody could I merge that?

Reply:
Okay, I just checked and even if you reorder the lines of this test to put `shutdownController.abort()` *before* the call to `dispatcher.retrieveCommit`, the test works fine against go-ipfs.  So it seems like this is a js-ipfs specific bug that they don't handle already-activated abortSignals.  Can we check the test configuration and put the sleep under an if-statement so it only runs if we're in js-ipfs mode, but doesn't sleep if we're in go-ipfs mode?

Might also want to increase the sleep to 1 second just to make the test less likely to be flaky on a machine with different performance characteristics.

Reply:
There are two ways to check if the test runs with js-ipfs. One is to look at an environment variable, another is to check functions specific to IPFS node. I believe, using an env variable creates unnecessary connascence. Added a check using ipfs-node-specific `preload` method.